### ISSUE TITLE
Concurrent access

### ISSUE THREAD

**Issue by nikoncode** [hazelcast/hazelcast-tomcat-sessionmanager#34]
I have a lot of concurrent calls inside my application and I receive the following logs:
```
Nov 24, 2017 6:35:16 AM com.hazelcast.session.HazelcastSessionManager findSession
INFO: No Session found for:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:16 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-7 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:16 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager findSession
INFO: No Session found for:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager findSession
INFO: No Session found for:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-3 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-1 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-7 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-7 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-7 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-7 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-7 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-1 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-7 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-3 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-3 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-7 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager findSession
INFO: No Session found for:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-7 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-3 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:17 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:18 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-3 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:18 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-3 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:18 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-7 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:18 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-3 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:18 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:18 AM com.hazelcast.session.HazelcastSessionManager findSession
INFO: No Session found for:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:18 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-3 committed key:ADC7938F491576F2703AFEA344EB0A3F
Nov 24, 2017 6:35:18 AM com.hazelcast.session.HazelcastSessionManager commit
INFO: Thread name:http-bio-8080-exec-6 committed key:ADC7938F491576F2703AFEA344EB0A3F
```

As you can see I always have the same session id, but for some reason session manager cannot find it.

I think the root cause is the following code: 
```
this.sessionMap.remove(id);
this.sessionMap.set(id, hazelcastSession);
```

Can somebody explain me why we cannot just execute set without remove to prevent concurrency issue?

If it's required how I can prevent problem like this.

**Comment by nikoncode** [comment]
Can we use `set` without remove or `put` instead of `remove` & `set` operation?

**Comment by mesutcelik** [comment]
similar to #42

**Comment by alparslanavci** [comment]
Fixed by #61.
